From 2e7fd9854f96ace0e3121e6ccc72c4cf4d32f143 Mon Sep 17 00:00:00 2001
From: Pieter-Jan Arts <pj.arts@uselab.com>
Date: Thu, 22 Aug 2024 14:46:25 +0200
Subject: [PATCH 1/2] Prevent href content from being executed

---
 js/src/carousel.js | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/js/src/carousel.js b/js/src/carousel.js
index aed44e1..d90086d 100644
--- a/js/src/carousel.js
+++ b/js/src/carousel.js
@@ -541,14 +541,15 @@ class Carousel {
 
   static _dataApiClickHandler(event) {
     const selector = Util.getSelectorFromElement(this)
-
     if (!selector) {
+      event.preventDefault()
       return
     }
 
     const target = $(selector)[0]
 
     if (!target || !$(target).hasClass(CLASS_NAME_CAROUSEL)) {
+      event.preventDefault()
       return
     }
 
@@ -556,13 +557,20 @@ class Carousel {
       ...$(target).data(),
       ...$(this).data()
     }
+
     const slideIndex = this.getAttribute('data-slide-to')
 
     if (slideIndex) {
       config.interval = false
     }
 
-    Carousel._jQueryInterface.call($(target), config)
+    try {
+      Carousel._jQueryInterface.call($(target), config)
+    } catch (error) {
+      console.log(error)
+      event.preventDefault()
+      return
+    }
 
     if (slideIndex) {
       $(target).data(DATA_KEY).to(slideIndex)
-- 
2.43.0

